### ISSUE TITLE
Consider disabling emacs-lisp-checkdoc in org-edit-special buffers by default

### ISSUE THREAD

**Issue by jody-frankowski** [flycheck/flycheck#1623]
## Bug description
Flycheck doesn't disable emacs-lisp-checkdoc for org-mode source blocks.

## Steps to reproduce
1. Write an emacs-lisp code block in a .org file such as this one:

```
Disable flycheck's emacs-lisp-checkdoc for org-edit-special buffers:

#+BEGIN_SRC emacs-lisp
(add-hook 'org-src-mode-hook
                  (lambda ()
                   (when (eq major-mode 'emacs-lisp-mode)
                     (setq flycheck-disabled-checkers '(emacs-lisp-checkdoc)))))
#+END_SRC
```
2. Press `C-c '` over the source block
3. Notice the flycheck errors

## Expected behavior
Flycheck doesn't report the emacs-lisp-checkdoc because small emacs-lisp code chunks exported from org files are not meant to be full-fledged files.

## My Personal FIx
```
(add-hook 'org-src-mode-hook
                 (lambda ()
                   (when (eq major-mode 'emacs-lisp-mode)
                     (setq flycheck-disabled-checkers '(emacs-lisp-checkdoc)))))
```

**Comment by fmdkdd** [comment]
You are right that the checkdoc warnings are not relevant for Elisp snippets.  They are not relevant for most non-package Elisp files either.

It would be helpful to find a way to enable checkdoc only for relevant Elisp buffers, but this fix is too narrow.

Thank you for providing one though, in case anyone has the same issue.  I'm closing as wontfix for now, but if you have a better solution, we can still discuss it.
